command/agent/retry_join.go

```diff
@@ -30,6 +35,57 @@ type DiscoverInterface interface {
 	Names() []string
 }
 
+// discoverProxy proxies retryJoiner discovery requests to either go-netaddrs or go-discover
+//
+// Addresses are first discovered using go-netaddrs, but if no addresses are able to be disocovered,
+// fall back to go-autodiscover
+type discoverProxy struct {
```

Comment:
`autoDiscover` may be a better name; `*Proxy` feels a bit non-specific.

Reply:
Agreed.